src/ringmesh/io/geomodel/io_vtk.hpp

```diff
@@ -75,58 +51,235 @@ namespace
+         * @brief write the vertices coordinates
+         * @details the vertices being written are the ones on the GeoModelMesh
+         * @param[in] geomodel the GeoModel being exported
+         * @param[in,out] out the ofstream being completed
+         */
+        void write_vertices_attributes(const GeoModel3D& geomodel,
+                std::ofstream& out) {
+            auto& attribute_manager = geomodel.mesh.vertices.attribute_manager();
+            out << "POINT_DATA " << geomodel.mesh.vertices.nb() << EOL;
+            GEO::vector< std::string > attribute_names;
+            attribute_manager.list_attribute_names(attribute_names);
+            for(auto attribute_index : range(attribute_manager.nb()) ){
+                std::string attribute_name = attribute_names[attribute_index];
+                if( GEO::AttributeBase< double >::is_defined(attribute_manager,
+                            attribute_name) ) {
+                    GEO::Attribute< double > cur_attribute(attribute_manager,
+                            attribute_name);
+                    auto dimension = cur_attribute.dimension();
+                    if( dimension == 1 ) {
+                        out << "SCALARS " << attribute_name <<  " double 1" << EOL;
+                        out << "LOOKUP_TABLE default" << EOL;
+                    }
+                    if( dimension > 1 ) {
+                        out << "VECTORS " << attribute_name <<  " double" << EOL;
+                    }
+                    for( auto index : range( cur_attribute.size() ) ) {
+                        for(auto dim : range(dimension) ) {
+                            out << cur_attribute[dimension*index + dim] << SPACE;
+                        }
+                        out << EOL;
+                    }
+                }
+                else {
+                    Logger::warn("I/O", "Attribute ", attribute_name, "is not a double ",
+                            "attribute and will not be exported");
+                }
             }
             out << EOL;
+        }
 
-            index_t total_corners = ( 4 + 1 ) * mesh.cells.nb_tet()
-                                    + ( 5 + 1 ) * mesh.cells.nb_pyramid()
-                                    + ( 6 + 1 ) * mesh.cells.nb_prism()
-                                    + ( 8 + 1 ) * mesh.cells.nb_hex();
-            out << "CELLS " << mesh.cells.nb_cells() << SPACE << total_corners
+        /*!
+         * @brief write the elements
+         * @details in VTK, elements are corner vertices, edges, polygons and
+         * cells in the RINGMesh point of view. All these elements are being written.
+         * @param[in] geomodel the GeoModel being exported
+         * @param[in,out] out the ofstream being completed
+         */
+        void write_elements(const GeoModel3D& geomodel, std::ofstream& out) {
+            const auto& geomodel_mesh = geomodel.mesh;
+            index_t total_corners = ( 4 + 1 ) * geomodel_mesh.cells.nb_tet()
```

Comment:
what are these corners? Add a comment to explain difference with RINGMesh Corners